src/lakefs_spec/client_helpers.py

```diff
@@ -63,3 +64,22 @@ def revert(client: LakeFSClient, repository: str, branch: str, parent_number: in
     client.branches_api.revert_branch(
         repository=repository, branch=branch, revert_creation=revert_creation
     )
+
+
+def rev_parse(
+    client: LakeFSClient,
+    repository: str,
+    ref: str,
+    parent: int = 0,
+) -> str:
+    if parent < 0:
+        raise ValueError(f"Parent cannot be negative, got {parent}")
+    try:
+        revisions = client.refs_api.log_commits(
+            repository=repository, ref=ref, limit=True, amount=2 * (parent + 1)
+        ).results
+    except NotFoundException:
+        raise RuntimeError(
+            f"{ref!r} does not match any revision in lakeFS repository {repository!r}"
+        )
+    return revisions[parent].id
```

Comment:
Can you move this return into the `try` block?

Also, from some local exploration it looks like `amount` is the _maximum_ amount of results - so it is possible to produce an `IndexError` here if you request a `parent` of an existing commit that has fewer than `parent` ancestors.

Suggestion: add an `if len(revisions) < parents` check which throws a `ValueError("cannot fetch revision {ref}~{parent}: {ref} only has {len(revisions} parents")`  (or something like that) if the number of ancestors is lower than requested.